### PR TITLE
DCA-110: Improve reading yml config files

### PR DESCRIPTION
**Pull request by mmizin**

### CODE REVIEW

File: app/selenium_ui/conftest.py [hunk 1]
```diff
@@ -48,17 +49,6 @@ def __get_current_results_dir():
         pass
 
 
-def application_url(yml_file_path: Path):
-    with yml_file_path.open(mode='r') as fs:
-        jira_yaml = yaml.load(fs, Loader=yaml.FullLoader)
```
Comment (astashys) [inline app/selenium_ui/conftest.py]: you deleted yml file do not forget delete unused imports (`ctrl+alt+o` in pycharm)

File: app/util/data_preparation/confluence/api.py [hunk 2]
```diff
@@ -5,15 +5,10 @@
...
 
-def admin_credentials():
-    # TODO extract paths to project_paths
-    with open(Path(__file__).parents[3] / "confluence.yml", 'r') as file:
-        jira_yaml = yaml.load(file, Loader=yaml.FullLoader)
-        return jira_yaml['settings']['env']['admin_login'], jira_yaml['settings']['env']['admin_password']
 
-
-USER, PASSWORD = admin_credentials()
+USER, PASSWORD = CONFLUENCE_SETTINGS.admin_login, CONFLUENCE_SETTINGS.admin_password
```
Comment (astashys) [inline app/util/data_preparation/confluence/api.py]: try to rewrite this class do not import util.conf but take user and password as input parameters in the constructor, see ApiJira for reference

File: app/util/conf.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        self.hostname = obj['settings']['env']['application_hostname']
+        self.protocol = obj['settings']['env']['application_protocol']
+        self.port = obj['settings']['env']['application_port']
+        self.postfix = obj['settings']['env']['application_postfix'] or ""
+        self.admin_login = obj['settings']['env']['admin_login']
+        self.admin_password = obj['settings']['env']['admin_password']
+        self.concurrency = obj['settings']['env']['concurrency']
+
+
+JIRA_SETTINGS = JiraSettings()
```
Comment (astashys) [inline app/util/conf.py]: What do you think to make these vars be shorter just `JIRA` and `CONFLUENCE`?

File: app/util/conf.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        self.admin_login = obj['settings']['env']['admin_login']
+        self.admin_password = obj['settings']['env']['admin_password']
+        self.concurrency = obj['settings']['env']['concurrency']
+
+    @property
+    def server_url(self):
+        return f'{self.protocol}://{self.hostname}:{self.port}{self.postfix}'
+
+
+class ConfluenceSettings(JiraSettings):
```
Comment (astashys) [inline app/util/conf.py]: please remove inheritance from JiraSettings

File: app/util/conf.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    def __init__(self):
+        super().__init__()
```
Comment (astashys) [inline app/util/conf.py]: remove this also please

File: app/util/data_preparation/confluence/api.py [hunk 6]
```diff
@@ -5,15 +5,10 @@
 import requests
 import yaml
```
Comment (astashys) [inline app/util/data_preparation/confluence/api.py]: remove this unused import please

File: app/util/data_preparation/confluence/prepare-data.py [hunk 7]
```diff
@@ -4,19 +4,13 @@
 import yaml
```
Comment (astashys) [inline app/util/data_preparation/confluence/prepare-data.py]: please remove yaml import, also sys import

File: app/util/data_preparation/jira/prepare-data.py [hunk 8]
```diff
@@ -2,11 +2,10 @@
 import string
 import sys
```
Comment (astashys) [inline app/util/data_preparation/jira/prepare-data.py]: please remove this import

File: app/util/data_preparation/jira/prepare-data.py [hunk 9]
```diff
@@ -136,14 +128,10 @@ def __get_software_project_keys(jira_api):
...
-    # TODO consider getting server url from jira.yml see get_perf_users_count()
-    url = sys.argv[1]
+    url = JIRA_SETTINGS.server_url
     print("Server url: ", url)
 
-    # TODO consider reading jira.yml only once
-    with JIRA_YML.open(mode='r') as file:
-        jira_yaml = yaml.load(file, Loader=yaml.FullLoader)
-        user, password = jira_yaml['settings']['env']['admin_login'], jira_yaml['settings']['env']['admin_password']
+    user, password = JIRA_SETTINGS.admin_login, JIRA_SETTINGS.admin_password
```
Comment (astashys) [inline app/util/data_preparation/jira/prepare-data.py]: redundant variables, just put `ApiJira(url, JIRA_SETTINGS.admin_login, JIRA_SETTINGS.admin_password)`

File: app/util/project_paths.py [hunk 10]
```diff
@@ -51,5 +51,4 @@ def __get_confluence_yml():
 JIRA_DATASET_USERS = __get_jira_dataset_users()
 JIRA_DATASET_ISSUES = __get_jira_dataset_issues()
 JIRA_DATASET_PROJECT_KEYS = __get_jira_projects_key()
-
```
Comment (astashys) [inline app/util/project_paths.py]: it would be nice to have this space to separate Jira and Confluence vars

File: app/util/conf.py (hunk 3 above)
Comment (astashys) [reply app/util/conf.py]: makes sense

Comment (astashys) [reply app/util/conf.py]: but JIRA_YML_CONFIG is too large as for me, currently, we do not have other settings except yml, maybe JIRA_CONFIG or JIRA_CONF is enough

File: app/selenium_ui/conftest.py (hunk 1 above)
Comment (mmizin) [reply app/selenium_ui/conftest.py]: Thx for the shortcut, very useful!

File: app/util/data_preparation/confluence/api.py (hunk 2 above)
Comment (mmizin) [reply app/util/data_preparation/confluence/api.py]: Good idea, I'll rewrite

File: app/util/conf.py (hunk 3 above)
Comment (mmizin) [reply app/util/conf.py]: If shorten Confluence, it will be looking like CONF_CONF.
Seems to be nice )))

**Comment by astashys** [comment]
looks good to me

**Pull request merged**